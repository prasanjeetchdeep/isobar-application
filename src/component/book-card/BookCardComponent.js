import React, {useState} from "react";
import { Col, Card } from "react-bootstrap";
import "./BookCardComponent.modules.scss";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faPlusCircle, faTimesCircle, faClock, faCalendar } from '@fortawesome/free-solid-svg-icons'

const BookCardComponent = ({ id, image, title, description, author, duration, publishDate, isAddedtoCart, onButtonClick}) => {
  const [showButton, setShowButton] = useState(false);
  
  return (
    <Col lg={3} md={4} sm={6} xs={12} style={{ marginTop: 15 }}>
      <Card onMouseEnter={() => {setShowButton(true)}} onMouseLeave={() => {setShowButton(false)}} style={{ boxShadow: "2px 2px 10px #e2e2e2", cursor: "pointer", position: "relative" }}>
      {showButton && !isAddedtoCart && <span className="add-to-cart" onClick={() => onButtonClick(id) }><FontAwesomeIcon icon={faPlusCircle} />  Add to cart</span>} 
      {showButton && isAddedtoCart && <span className="remove-from-cart" onClick={() => onButtonClick(id) }><FontAwesomeIcon icon={faTimesCircle} /> Remove from cart</span>} 
        <div className="flexbox" style={{ placeContent: "center", padding: 5}}>
          <img  src={image} alt={title} style={{ height: 120, opacity: showButton ? 0.7 : 1 }} />
        </div>

        <Card.Body>
          <Card.Title>{title}</Card.Title>
          <Card.Text>{author}</Card.Text>
          <Card.Text>{description}</Card.Text>
          
          <Card.Text>
            <span className="bottom-details duration-section"><FontAwesomeIcon icon={faClock} /> {duration}</span> 
            <span className="bottom-details publish-date-section"><FontAwesomeIcon icon={faCalendar} /> {publishDate}</span></Card.Text>
           </Card.Body>
      </Card>
    </Col>
  );
};

export default BookCardComponent;
