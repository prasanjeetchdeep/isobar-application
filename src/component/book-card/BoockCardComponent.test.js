import React from "react";
import { render } from "@testing-library/react";
import BookCardComponent from "./BookCardComponent";

const book = [
    {
      id: 1,
      name: "React - basics",
      description: "This course is going to take you through basics of React.",
      author: "James White",
      publishDate: "12/03/2019",
      duration: "00:03:56",
      image: "https://cdn.auth0.com/blog/react-js/react.png"
    }
  ];
  const noop = () => {};

it("should render the Book Card if the book is added in the cart", () => {
    const { asFragment } = render(
    <BookCardComponent
    book={book}
    isAddedtoCart={true}
    onButtonClick={noop}
     />
    );
    expect(asFragment()).toMatchSnapshot();
});

it("should render the Book Card if the book is not added in the cart", () => {
    const { asFragment } = render(
    <BookCardComponent
    book={book}
    isAddedtoCart={false}
    onButtonClick={noop}
     />
    );
    expect(asFragment()).toMatchSnapshot();
});