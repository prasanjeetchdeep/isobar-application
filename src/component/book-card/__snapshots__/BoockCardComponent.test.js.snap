// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Book Card if the book is added in the cart 1`] = `
<DocumentFragment>
  <div
    class="col-lg-3 col-md-4 col-sm-6 col-12"
    style="margin-top: 15px;"
  >
    <div
      class="card"
      style="box-shadow: 2px 2px 10px #e2e2e2; cursor: pointer; position: relative;"
    >
      <div
        class="flexbox"
        style="place-content: center; padding: 5px;"
      >
        <img
          style="height: 120px; opacity: 1;"
        />
      </div>
      <div
        class="card-body"
      >
        <div
          class="bookcard-title card-title h5"
        />
        <p
          class="card-text"
        />
        <p
          class="card-text"
        />
        <p
          class="card-text"
        >
          <span
            class="bottom-details duration-section"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-clock fa-w-16 "
              data-icon="clock"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
                fill="currentColor"
              />
            </svg>
             
          </span>
          <span
            class="bottom-details publish-date-section"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-calendar fa-w-14 "
              data-icon="calendar"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                fill="currentColor"
              />
            </svg>
             
          </span>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render the Book Card if the book is not added in the cart 1`] = `
<DocumentFragment>
  <div
    class="col-lg-3 col-md-4 col-sm-6 col-12"
    style="margin-top: 15px;"
  >
    <div
      class="card"
      style="box-shadow: 2px 2px 10px #e2e2e2; cursor: pointer; position: relative;"
    >
      <div
        class="flexbox"
        style="place-content: center; padding: 5px;"
      >
        <img
          style="height: 120px; opacity: 1;"
        />
      </div>
      <div
        class="card-body"
      >
        <div
          class="bookcard-title card-title h5"
        />
        <p
          class="card-text"
        />
        <p
          class="card-text"
        />
        <p
          class="card-text"
        >
          <span
            class="bottom-details duration-section"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-clock fa-w-16 "
              data-icon="clock"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
                fill="currentColor"
              />
            </svg>
             
          </span>
          <span
            class="bottom-details publish-date-section"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-calendar fa-w-14 "
              data-icon="calendar"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 192h424c6.6 0 12 5.4 12 12v260c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V204c0-6.6 5.4-12 12-12zm436-44v-36c0-26.5-21.5-48-48-48h-48V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H160V12c0-6.6-5.4-12-12-12h-40c-6.6 0-12 5.4-12 12v52H48C21.5 64 0 85.5 0 112v36c0 6.6 5.4 12 12 12h424c6.6 0 12-5.4 12-12z"
                fill="currentColor"
              />
            </svg>
             
          </span>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
